### PR TITLE
Updates ISIS to use the unified ale format

### PR DESCRIPTION
**Pull request by krlberry**
## Description
Updates ISIS to use the new unified ale isd format. This PR almost entirely @Kelvinrr's changes from PR #3839, which I reverted in PR #3888 to fix failures occurring in dev, with a small update from me to get tests passing.

## Related Issue
Updates ALE format isd format used in ISIS to the unified "ale" format, from the "isis" format. 

## Motivation and Context
Updates ALE format isd format used in ISIS to the unified "ale" format, from the "isis" format. 

## How Has This Been Tested?
Did a full run of tests locally. All tests pass.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/Spice/Spice.cpp [hunk 1]
```diff
@@ -965,8 +965,7 @@ namespace Isis {
 
 
   /**
-   * This returns the PvlObject that stores all of the requested Naif data
-   *   and can be a replacement for furnishing text kernels.
+   * This returns the PvlObject that stores all of the requested Naif data  *   and can be a replacement for furnishing text kernels.
```
Comment (jessemapel) [inline isis/src/base/objs/Spice/Spice.cpp]: Why is this change here?

File: isis/src/base/objs/SpiceRotation/unitTest.cpp [hunk 2]
```diff
@@ -1,890 +1,350 @@
...
-//TODO test loadPCFromSpice() and loadPCFromTable() methods
-//TODO see end of unit test for exceptions that need to be tested
+using namespace std;
 
 int main(int argc, char *argv[]) {
   Preference::Preferences(true);
 
   cout << setprecision(8);
-  cout << "Unit test for SpiceRotation" << endl;
+  cout << "Unit test for SpicePosition" << endl;
```
Comment (jessemapel) [inline isis/src/base/objs/SpiceRotation/unitTest.cpp]: Why is all of this moved into the SpiceRotation unit test?

Comment (krlberry) [reply isis/src/base/objs/SpiceRotation/unitTest.cpp]: Woah. I don't know how this happened, thanks for catching this. I'll fix it.

File: isis/src/base/objs/Spice/Spice.cpp (hunk 1 above)
Comment (krlberry) [reply isis/src/base/objs/Spice/Spice.cpp]: This was unintentional, thanks for catching it!

**Review by jessemapel — COMMENTED** [review]
Changes look good but all of the ThreeImageNetwork tests are segfaulting in CI

```
The following tests FAILED:
	1601 - ThreeImageNetwork.FunctionalTestAutoseedDefault (SEGFAULT)
	1602 - ThreeImageNetwork.FunctionalTestAutoseedCnetInput (SEGFAULT)
	1603 - ThreeImageNetwork.FunctionalTestAutoseedDeffiles (SEGFAULT)
	1604 - ThreeImageNetwork.FunctionalTestCnetwinnowDefault (SEGFAULT)
	1605 - ThreeImageNetwork.FunctionalTestCnetstatsDefault (SEGFAULT)
	1606 - ThreeImageNetwork.FunctionalTestCnetstatsImageStats (SEGFAULT)
	1607 - ThreeImageNetwork.FunctionalTestCnetstatsPointStats (SEGFAULT)
	1608 - ThreeImageNetwork.FunctionalTestCnetstatsCubeFilter (SEGFAULT)
	1609 - ThreeImageNetwork.FunctionalTestCnetstatsPointFilter (SEGFAULT)
	1610 - ThreeImageNetwork.FunctionalTestCnetstatsInvalidDefFile (SEGFAULT)
	1611 - ThreeImageNetwork.FunctionalTestCnetbin2pvlDefault (SEGFAULT)
	1612 - ThreeImageNetwork.FunctionalTestCnetpvl2binDefault (SEGFAULT)
	1613 - ThreeImageNetwork.FunctionalTestFindImageOverlapsNoOverlap (SEGFAULT)
	1614 - ThreeImageNetwork.FunctionalTestCnetcheckCamera (SEGFAULT)
	1615 - ThreeImageNetwork.FunctionalTestCnetcheckNoPoints (SEGFAULT)
	1616 - ThreeImageNetwork.FunctionalTestCnetcheckIgnore (SEGFAULT)
Errors while running CTest
```

**Comment by jessemapel** [comment]
@kberryUSGS Any progress on the test failures?

**Review by acpaquette — COMMENTED** [review]
Overall looks good! Small change that will be needed once all of the ALE PRs are merged

File: isis/src/base/objs/SpiceRotation/SpiceRotation.cpp [hunk 3]
```diff
@@ -429,32 +429,32 @@ namespace Isis {
...
         std::vector<double> quat = {it->at(0).get<double>(), it->at(1).get<double>(), it->at(2).get<double>(), it->at(3).get<double>()};
         Quaternion q(quat);
         std::vector<double> CJ = q.ToMatrix();
         p_cache.push_back(CJ);
     }
 
-    if (isdRot["AngularVelocity"].size() != 0) {
-      for (auto it = isdRot["AngularVelocity"].begin(); it != isdRot["AngularVelocity"].end(); it++) {
+    if (isdRot["angular_velocity"].size() != 0) {
+      for (auto it = isdRot["angular_velocity"].begin(); it != isdRot["angular_velocity"].end(); it++) {
```
Comment (acpaquette) [inline isis/src/base/objs/SpiceRotation/SpiceRotation.cpp:446]: I changed the keyword for `angular_velocity` in ALE to output as `angular_velocities` to be a little more consistent since all of the other keywords are plural.

**Comment by krlberry** [comment]
@jessemapel @acpaquette Fixed the test errors. I didn't update the `angular_velocity` to `angular_velocities` since that wouldn't work with the current released version of ale that's currently being used with ISIS. As I see it, I could either make that change, but plan to leave the PR open until a new ale version is released, or have this PR merged and make that change in a future PR to work with the new ale release.

**Review by jessemapel — APPROVED** [review]
This looks ready to go now.
